torch/_dynamo/utils.py

```diff
@@ -1038,6 +1051,11 @@ def get_fake_value(node, tx):
     def visit(n: torch.fx.Node):
         return n.meta["example_value"]
 
+    if node.op == "call_function" and node.target in DYNAMO_SUPPORTED_COLLECTIVES:
```

Comment:
@voznesenskym the primary reason for this special case is that allreduce doesn't work with fake/meta tensors, so we can't actually run it.

but the secondary issue i ran into and also conveniently worked around is that the line below (map_arg, ..., visit) blows up on allreduce, since its processgroup input is an fx.node with no meta['example_input'].

I messed around with how to properly add that briefly and wasn't sure what the invariants are or whether a processgroup even should have its meta filled out.  The logical value there would just be the processgroup, afaiu.

Reply:
i don't mind taking another pass myself, so i can learn. i'm working on meta for allreduce first